src/app/core/services/basic-rundown-state.service.ts

```diff
@@ -0,0 +1,115 @@
+      private readonly connectionStatusObserver: ConnectionStatusObserver
+  ) {
+    this.registerEventConsumers()
+  }
+
+  private registerEventConsumers(): void {
+    const unsubscribeFromConnectionStatusEvents = this.registerConnectionStatusConsumers()
+    const unsubscribesFromRundownEvents = this.registerBasicRundownEventConsumers()
+    this.unsubscribeFromEvents = () => [
+        ...unsubscribesFromRundownEvents,
+        ...unsubscribeFromConnectionStatusEvents
+      ].forEach(unsubscribe => unsubscribe())
+  }
+
+  private registerConnectionStatusConsumers(): Unsubscribe[] {
+    return [
+      this.connectionStatusObserver.subscribeToConnectionStatus(this.connectionStatusHandler.bind(this))
+    ]
+  }
+
+  private connectionStatusHandler(isConnected: boolean): void {
+    if (!isConnected) {
+      return
+    }
+    this.resetBasicRundownSubject()
+  }
+
+  private resetBasicRundownSubject(): void {
+    console.log('[debug][RundownStateService]', 'Resetting basic rundowns')
+    this.fetchBasicRundowns()
+        .then(basicRundowns => this.basicRundownsSubject.next(basicRundowns))
+        .catch(error => console.error('[error]', 'Encountered error while fetching basic rundowns:', error))
+  }
+
+  private registerBasicRundownEventConsumers(): Unsubscribe[] {
+    return [
+      this.rundownEventObserver.subscribeToRundownActivation(this.createBasicRundownEventHandler(this.rundownActivationHandler.bind(this))),
+      this.rundownEventObserver.subscribeToRundownDeactivation(this.createBasicRundownEventHandler(this.rundownDeactivationHandler.bind(this))),
+      this.rundownEventObserver.subscribeToRundownDeletion(this.createBasicRundownEventHandler(this.rundownDeletionHandler.bind(this))),
+      // TODO: Add listener(s) for updating the modifiedAt attribute when we implement modifiedAt.
+      // TODO: Add listener(s) for when a rundown is created.
+    ]
+  }
+
+  private createBasicRundownEventHandler<EventType extends { rundownId: string }>(
+      handler: (basicRundown: BasicRundown, event: EventType) => BasicRundown[]
+  ): (event: EventType) => void {
+    return (event: EventType) => {
+      const basicRundowns = this.basicRundownsSubject.value
+      const newBasicRundowns: BasicRundown[] = basicRundowns.reduce(this.createBasicRundownReducer(handler, event), [])
+      this.basicRundownsSubject.next(newBasicRundowns)
+    }
+  }
+
+  private createBasicRundownReducer<EventType extends { rundownId: string }>(
+      handler: (basicRundown: BasicRundown, event: EventType) => BasicRundown[],
+      event: EventType
+  ): (basicRundowns: BasicRundown[], basicRundown: BasicRundown) => BasicRundown[] {
+    return (basicRundowns: BasicRundown[], basicRundown: BasicRundown) => {
+      if (basicRundown.id !== event.rundownId) {
+        return [...basicRundowns, basicRundown]
+      }
+      return [
+          ...basicRundowns,
+          ...handler(basicRundown, event)
+      ]
+    }
+  }
```

Comment:
I have removed this.

Reply:
>  but the resulting array might have a different shape.

If I understand it correctly then the reason that this might have a different shape is because your "DeleteHandler" returns an empty array?

That is a very hidden dependency on _one_ of your "handler" methods.

But as we talked about, you are trying to make one common tool that can handle all of your potential scenarios instead of having each scenario pick the tool that fit the scenario best.